### PR TITLE
Allow building host tests with code coverage

### PR DESCRIPTION
**Pull request by arkq**
### Problem

It seems that the code coverage for tests was functional in the past, but it's not working now.

### Changes

- Properly exclude tests from the coverage report
- Fix pigweed tests to work with google test backend (it is more restrictive in terms of `operator==` visibility than pigweed test light)

### Testing

Fragment of the report for `linux-x64-tests-coverage`:

image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 # Build System
 out/
+/examples/virtual-device-app/android/App/buildSrc/build/
```
Comment (andy31415) [inline .gitignore:13]: this change seems unrelated.

File: src/system/SystemPacketBuffer.h [hunk 2]
```diff
@@ -684,8 +686,7 @@ class DLL_EXPORT PacketBufferHandle
     }
 
     PacketBuffer * Get() const { return mBuffer; }
-
-    bool operator==(const PacketBufferHandle & aOther) const { return mBuffer == aOther.mBuffer; }
+    PacketBuffer * GetNext() const { return static_cast<PacketBuffer *>(mBuffer->next); }
```
[the remark this answers is not in the document]
Comment (andy31415) [reply src/system/SystemPacketBuffer.h:689]: I see it is marked private .... closing.

**Review by andy31415 — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (arkq) [reply .gitignore:13]: Yes, it's unrelated to the code coverage itself, but when building tests my git tree is dirty because of this directory being created. I can create a separate PR for that...

**Review by bzbarsky-apple — APPROVED** [review]
Rubber-stamping, since @andy31415 reviewed.
